assets/scripts/index.js

```diff
@@ -0,0 +1,88 @@
+const retrieveBooks = () => {
+  const bookList = document.querySelector('.books-list');
+  if (localStorage.getItem('books')) {
+    const books = JSON.parse(localStorage.getItem('books'))
+      .map(
+        (book) => `<li class="book-item" data-id="${book.id}">
+        <div class="book-item__info">
+          <h2 class="book-item__title">${book.title}</h2>
+          <p class="book-item__author">${book.author}</p>
+          <button class="removeFromShelfBtn" onclick="removeBook(${book.id})">
+            Remove
+          </button>
+        </div>  </li>`,
+      )
+      .join('');
+    bookList.innerHTML = books;
+  } else {
+    console.log('No books found');
+  }
+};
+
+window.onload = () => {
+  retrieveBooks();
+};
+const booksArray = [];
+
+const removeBook = (bookId) => {
+  const foundBook = JSON.parse(localStorage.getItem('books')).find(
+    (book) => book.id === bookId * 1,
+  );
+  const temp = JSON.parse(localStorage.getItem('books'));
+  const index = temp.map((object) => object.id).indexOf(foundBook.id);
+  temp.splice(index, 1);
+  localStorage.setItem('books', JSON.stringify(temp));
+  retrieveBooks();
+};
```

Comment:
- [ ] It is recommended to use the filter in this function and also notice that you should remove the book from booksArray or make the booksArray equal to temp.
```
const removeBook = (bookId) => {
  const temp = JSON.parse(localStorage.getItem('books')).filter(
    (book) => book.id !== bookId,
  );
  booksArray = temp; 
  localStorage.setItem('books', JSON.stringify(temp));
  retrieveBooks();
};
```